avocado/utils/vmimage.py

```diff
@@ -0,0 +1,375 @@
+        # In that case, URL is ready
+        if all_args:
+            return image_url.format(**self.initargs)
+
+        # Otherwise, we have to probe it externally
+        # Getting the requested version, if available
+        version = self.get_version()
+        # Getting the image URL
+        return self.get_image(version)
+
+    def get_version(self):
+        pattern = '^%s/$' % self.args['version']
+
+        parser = VMImageHtmlParser(pattern)
+        try:
+            parser.feed(urllib2.urlopen(self.url_versions).read())
+        except:
+            raise ImageProviderError('Cannot open %s' % self.url_versions)
+        if parser.versions:
+            resulting_versions = []
+            for version in parser.versions:
+                try:
+                    resulting_versions.append(int(version))
+                except ValueError:
+                    try:
+                        resulting_versions.append(float(version))
+                    except ValueError:
+                        resulting_versions.append(version)
+            return max(resulting_versions)
+        else:
+            raise ImageProviderError('Version %s not available at %s' %
+                                     (self.default_version, self.url_versions))
+
+    def get_image(self, version):
+        url_images = self.url_images.format(version=version,
+                                            build=self.args['build'],
+                                            arch=self.args['arch'])
+        image = self.image.format(version=version,
+                                  build=self.args['build'],
+                                  arch=self.args['arch'])
+        parser = VMImageHtmlParser(image)
+        try:
+            content = urllib2.urlopen(url_images).read()
+            parser.feed(content)
+        except:
+            raise ImageProviderError('Cannot open %s' % url_images)
+        if parser.versions:
+            return url_images + max(parser.versions)
+        else:
+            raise ImageProviderError("No images matching '%s' at '%s'" %
+                                     (image, url_images))
+
+    def remove(self):
+        """
+        Removes the VM Image
+        """
+        if self.path is not None:
+            os.remove(self.path)
+            self.path = None
+
+    @staticmethod
+    def _create_backed_image(source_image):
+        """
+        Creates a Qcow2 backed image using the original image
+        as backing file.
+        """
+        qemu_img = utils_path.find_command('qemu-img')
+
+        name, extension = os.path.splitext(source_image)
+        new_image = '%s-%s%s' % (name, str(uuid.uuid4()).split('-')[0],
+                                 extension)
+        cmd = '%s create -f qcow2 -b %s %s' % (qemu_img,
+                                               source_image,
+                                               new_image)
+        process.run(cmd)
+        return new_image
```

Comment:
I don't think we should get backed_image by default, because people might want to test and use custom snapshots. Yes, this is a useful method, but by default I'd return the basic file and if user asks for it we should create the backed image.

Reply:
Ok, `path` will have the path for the base original image and `snapshot` will have the path for the snapshot image.